enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -0,0 +1,851 @@
+  section for more details.
+
+## Proposal
+
+`istio-csr` will be managed by `cert-manager-operator` which can be enabled and configured by a user as and
+when required, using new custom resource which will be made available for configuring `istio-csr`.
+
+A new controller will be added to `cert-manager-operator` to manage and maintain the `istio-csr` deployment in
+desired state which can be monitored by user through the status sub-resource of the new custom resource.
+
+`istio-csr` controller will make use of static manifest templates for creating below resources required for
+successfully deploying `istio-csr` agent. Please refer `Implementation Details/Notes/Constraints` section for more details.
+* `certificate.cert-manager.io`
+* `clusterissuer.cert-manager.io`
+* `issuer.cert-manager.io`
+* `clusterrole.rbac.authorization.k8s.io`
+* `clusterrolebinding.rbac.authorization.k8s.io`
+* `deployment`
+* `service`
+* `servicemonitor.monitoring.coreos.com`
+* `namespace`
+* `role.rbac.authorization.k8s.io`
+* `rolebinding.rbac.authorization.k8s.io`
+* `serviceaccount`
+
+`istio-csr` CR object will be a singleton object, must be named `default` and an OpenShift user will have to
+create the same for enabling the deployment. `istio-csr` will be deployed in the namespace where the CR object is created.
+
+When `istiocsr` custom resource is created without defining `issuerref`, the operator will create a `ClusterIssuer` in the
+`cert-manager` namespace using a self-signed `Issuer`, which will act as a root CA. And the `ClusterIssuer` will be used to
+create an `Issuer` an intermediate CA, in the istio-csr deployed namespace and will used to obtain certificates for gRPC
+service made available for `OpenShift Service Mesh` and also for issuing the certificates requested by `OpenShift Service
+Mesh`. The `Issuer` in the istio-csr deployed namespace will be used for obtaining the signed certificates requested by 
+the `OpenShift Service Mesh`.
+
+Configurations made available in the spec of `istiocsr` custom resource are passed as command line arguments to `istio-csr`
+agent and updating these configurations would cause new rollout of the `istio-csr` deployment, which means a new pod will
+be created and old pod will terminated resulting in `istio-csr` restart.
+
+When an OpenShift user deletes `istiocsr` CR object `istio-csr` agent deployment will be deleted and all the
+other resources created will also be cleaned up, including the `clusterissuer` and `issuer` objects if created by the
+controller.
+
+A fork of [upstream](https://github.com/cert-manager/istio-csr) `istio-csr` will be created [downstream](https://github.com/openshift/cert-manager-istio-csr)
+for better version management.
+
+### Workflow Description
+
+- Enable istio-csr deployment
+  - An OpenShift user enables `istio-csr` by creating the new custom resource made available for configuring
```

Comment:
"Enables" signifies that istio-csr agent is already installed. I believe, "installs" would be the right term.